babel/src/test/java/org/apache/calcite/test/BabelTest.java

```diff
@@ -239,6 +239,47 @@ private void checkInfixCast(Statement statement, String typeName, int sqlType)
         "EXPR$0=10\n");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5087">[CALCITE-5087]
+   * Support bitwise operators</a>. */
+  @Test void testBitwiseOperator() {
+    final SqlValidatorFixture f = Fixtures.forValidator()
+        .withParserConfig(c -> c.withParserFactory(SqlBabelParserImpl.FACTORY))
+        .withOperatorTable(operatorTableFor(SqlLibrary.MYSQL));
+
+    f.withSql("select 1 | 1")
+        .ok()
+        .type("RecordType(BIGINT NOT NULL EXPR$0) NOT NULL");
+
+    f.withSql("select x'65' & x'77'")
+        .ok()
+        .type("RecordType(BIGINT NOT NULL EXPR$0) NOT NULL");
+
+    f.withSql("select x'65' ^^ 1")
```

Comment:
This seems wrong, the output type should be BINARY or VARBINARY.
Or perhaps shifts of BINARY values should be forbidden.